### PR TITLE
`std.os.linux.start_pie`: Add support for the new RELR relocation format.

### PR DESCRIPTION
**Pull request by alexrp**
Some helpful resources:

* https://maskray.me/blog/2021-10-31-relative-relocations-and-relr
* https://dram.page/p/relative-relocs-explained

### CODE REVIEW

**Comment by alexrp** [comment]
cc @kubkon

File: lib/std/os/linux/start_pie.zig [hunk 1]
```diff
@@ -266,4 +266,28 @@ pub fn relocate(phdrs: []elf.Phdr) void {
...
+    if (relr != 0) {
+        const relrs = @call(.always_inline, std.mem.bytesAsSlice, .{
+            elf.Relr,
+            @as([*]u8, @ptrFromInt(base_addr + relr))[0..sorted_dynv[elf.DT_RELRSZ]],
+        });
+        var current: [*]usize = undefined;
+        for (relrs) |r| {
+            if ((r & 1) == 0) {
+                current = @ptrFromInt(base_addr + r);
+                current[0] += base_addr;
```
Comment (alexrp) [inline lib/std/os/linux/start_pie.zig:280]: As I worked on the relocation stuff in this file, I did start to wonder: Should these actually be done with wrapping arithmetic in general?

Comment (Rexicon226) [reply lib/std/os/linux/start_pie.zig:280]: No, I don't think so. This would never overflow unless a bug.

File: lib/std/os/linux/start_pie.zig [hunk 2]
```diff
@@ -266,4 +266,28 @@ pub fn relocate(phdrs: []elf.Phdr) void {
...
+                current += 1;
+            } else {
+                // Skip the first bit; there are 63 locations in the bitmap.
```
Comment (Rexicon226) [inline lib/std/os/linux/start_pie.zig:283]: Just for future reference to anyone, see the explanation here:
https://reviews.llvm.org/D48247

```
  // The bitmap entries must have 1 in the least significant bit. The assumption
  // here is that an address cannot have 1 in lsb. Odd addresses are not
  // supported.
```

**Review by Rexicon226 — APPROVED** [review]
This looks good and correct to me.

**Pull request merged**